### ISSUE TITLE
"RTNETLINK answers: Operation not supported" when running traffic-control labs in MacOS

### ISSUE THREAD

**Issue by marcelofernandez** [KatharaFramework/Kathara#265]
Hi!

I was preparing some traffic control labs for our students and I found this error when running any of the [Traffic Control labs](https://github.com/KatharaFramework/Kathara-Labs/tree/main/tutorials/traffic-control) on Mac OS (M1/ARM64 architecture). This doesn't happen on a Linux/x86_64 host.

![image](https://github.com/KatharaFramework/Kathara/assets/1588034/48feb656-4dc4-43df-8eae-ebc3d6788cd1)

I think the containers should be given the `--cap-add=NET_ADMIN` parameter when running the `docker run` command to avoid this issue, but I'm not sure how to add/define it (nor either I know if it would fix it, anyway).

Could you help me on this one?

Thank you!

**Comment by marcelofernandez** [comment]
Ok, I read some bits of the source code, and it seems that [those process capabilities are passed to the docker runner only when Kathara is being executed under privileged mode](https://github.com/KatharaFramework/Kathara/blob/main/src/Kathara/manager/docker/DockerMachine.py#L287). 

Then I was able to run the same lab with the `--privileged` parameter as root to be able to get those NET_ADMIN capabilities ([among others](https://github.com/KatharaFramework/Kathara/blob/main/src/Kathara/model/Machine.py#L25)), but the error still persists:

![image](https://github.com/KatharaFramework/Kathara/assets/1588034/a124d57d-e58c-4020-aecb-3c7ad689d506)

(Now the error message is `RTNETLINK answers: Operation not supported`) 🤷 

So, I'm not sure how to fix this one... thank you!

**Comment by Skazza94** [comment]
Hi @marcelofernandez,
thanks for the issue and the detailed troubleshooting.

We also managed to replicate the problem on Apple silicon. 

Apparently, the Docker VM on Apple silicon does not have the `netem` kernel module enabled. We opened an issue in the Docker repository https://github.com/docker/for-mac/issues/7138 to notify the problem. 

Let's see if they will enable the kernel module in future releases.

Just for completeness, `netem` is enabled in the Docker VM on Intel silicon.

We found this repository: https://github.com/lukaszlach/docker-tc which should bypass the limitation, but we did not test it and we do not know if it works on M1. 

Mariano.
